glide.yaml

```diff
@@ -0,0 +1,268 @@
+package: k8s.io/federation
+import:
+- package: cloud.google.com/go
+  subpackages:
+  - compute/metadata
+- package: github.com/aws/aws-sdk-go
+  subpackages:
+  - aws
+  - aws/request
+  - aws/session
+  - service/route53
+- package: github.com/coreos/etcd
+  subpackages:
+  - client
+- package: github.com/davecgh/go-spew
+  subpackages:
+  - spew
+- package: github.com/go-openapi/spec
+- package: github.com/gogo/protobuf
+  subpackages:
+  - proto
+- package: github.com/golang/glog
+- package: github.com/miekg/coredns
+  subpackages:
+  - middleware/etcd/msg
+- package: github.com/onsi/ginkgo
+- package: github.com/onsi/gomega
+- package: github.com/pborman/uuid
+- package: github.com/prometheus/client_golang
+  subpackages:
+  - prometheus
+- package: github.com/spf13/cobra
+  subpackages:
+  - doc
+- package: github.com/spf13/pflag
+- package: github.com/stretchr/testify
+  subpackages:
+  - assert
+  - require
+- package: golang.org/x/net
+  subpackages:
+  - context
+- package: golang.org/x/oauth2
+  subpackages:
+  - google
+- package: google.golang.org/api
+  subpackages:
+  - compute/v1
+  - dns/v1
+  - googleapi
+- package: gopkg.in/gcfg.v1
+- package: gopkg.in/yaml.v2
+- package: k8s.io/api
+  subpackages:
+  - apps/v1beta2
+  - autoscaling/v1
+  - batch/v1
+  - core/v1
+  - extensions/v1beta1
+- package: k8s.io/apimachinery
+  subpackages:
+  - pkg/api/errors
+  - pkg/api/meta
+  - pkg/api/resource
+  - pkg/apimachinery/announced
+  - pkg/apimachinery/registered
+  - pkg/apis/meta/internalversion
+  - pkg/apis/meta/v1
+  - pkg/conversion
+  - pkg/fields
+  - pkg/labels
+  - pkg/runtime
+  - pkg/runtime/schema
+  - pkg/runtime/serializer
+  - pkg/selection
+  - pkg/types
+  - pkg/util/errors
+  - pkg/util/intstr
+  - pkg/util/net
+  - pkg/util/rand
+  - pkg/util/runtime
+  - pkg/util/sets
+  - pkg/util/uuid
+  - pkg/util/validation
+  - pkg/util/validation/field
+  - pkg/util/wait
+  - pkg/util/yaml
+  - pkg/watch
+- package: k8s.io/apiserver
+  subpackages:
+  - pkg/admission
+  - pkg/endpoints/request
+  - pkg/registry/generic
+  - pkg/registry/generic/registry
+  - pkg/registry/rest
+  - pkg/server
+  - pkg/server/filters
+  - pkg/server/healthz
+  - pkg/server/options
+  - pkg/server/storage
+  - pkg/storage
+  - pkg/storage/names
+  - pkg/storage/storagebackend
+  - pkg/util/flag
+  - pkg/util/logs
+  - pkg/util/webhook
+- package: k8s.io/client-go
+  subpackages:
+  - discovery
+  - discovery/fake
+  - dynamic
+  - kubernetes
+  - kubernetes/scheme
+  - listers/core/v1
+  - rest
+  - testing
+  - tools/cache
+  - tools/clientcmd
+  - tools/clientcmd/api
+  - tools/leaderelection
+  - tools/leaderelection/resourcelock
+  - tools/record
+  - util/cert
+  - util/cert/triple
+  - util/flowcontrol
+  - util/workqueue
+- package: k8s.io/kube-openapi
+  subpackages:
+  - pkg/common
+- package: k8s.io/kubernetes
+  repo: git@github.com:marun/kubernetes
```

Comment:
This means that we are building an older date federation code, which misses some recent stuff. Can this pose any risk?
How I see it as, that we can pull all k/k/f commits on k/f and change the paths to compile latest packages from k/f instead, but I suspect that there might be discrepancies (Our same problem of skew between k/k/staging and k/api, k/apiserver etc). This is comment is not to block us, but just for thoughts!

Reply:
I'm told that we need to use the repos updated from staging (e.g. k/apimachinery), and those repos are only updated periodically.  This means we can't use HEAD of k/k, and need instead to use the k/k commit that HEAD of the staging repos refer to.  I'm assuming the lag is temporary, and that we will be able to converge faster in the future.

As per the included README, there is also the issue of the k/k generated files for openapi.  I think these files will be difficult or even impossible to generate from k/f, and to me that implies the need to maintain a fork of k/k that includes the generated files until k/f no longer depends on them (i.e. the fcp defines its own federation-specific api rather than using the kube api).

Reply:
Ack!